sympy/printing/tests/test_repr.py

```diff
@@ -101,12 +101,11 @@ def test_Rational():
 
 
 def test_Float():
-    sT(Float('1.23', prec=3), "Float('1.22998', prec=3)")
```

Comment:
In my view, it's better to always display in terms of binary precision to maintain the characteristics of the original Float. For example, we should print `Float('XXXX', precision = YYY)` if the Float was originally created with binary precision.

According to me(as per Sympy 1.0) , if I create : `Float('1.23', prec=3)` , I would expect the same to be printed out , because I explicitly mentioned `3` decimal digits.I don't see why it should be printed out as something different.But then again ,I may be incorrect in this.

I think that the value is stored correctly because both `a._mpf_` and `eval(srepr(a))._mpf_` are same.

Do you think it would be better if `srepr` prints with `dps=YY` when original Float is created with decimal precision and `precision=YY` for all other cases?